autoexec.sh

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env bash
+
+# Environment
+ENIGMA_INSTALL_DIR=${ENIGMA_INSTALL_DIR:=$HOME/enigma-bbs}
+AUTOEXEC_LOGFILE="$ENIGMA_INSTALL_DIR/logs/autoexec.log"
+
+# Mise en place
+~/.local/bin/mise activate bash >> bash
+
+# Environment Versions
+ENIGMA_NODE_VERSION=${ENIGMA_NODE_VERSION:=$(toml get --toml-path=$ENIGMA_INSTALL_DIR/mise.toml tools.node)}
+ENIGMA_PYTHON_VERSION=${ENIGMA_PYTHON_VERSION:=$(toml get --toml-path=$ENIGMA_INSTALL_DIR/mise.toml tools.python)}
+
+# Validate Environment
+DEPENDENCIES_VALIDATED=1
+
+echo "$(date '+%Y-%m-%d %H:%M:%S') - START:" | tee -a $AUTOEXEC_LOGFILE
```

Comment:
These echo's with dates and such could be turned into a function similar to the other script. You _could_ actually share them between the two by sourcing them into both